src/analysis/analysis-service.js

```diff
@@ -74,21 +75,52 @@ AnalysisFactory.prototype._getAnalysisAttributesFromAnalysisDefinition = functio
   return _.omit(_.extend(analysisDefinition, analysisDefinition.params, sourceNodes), 'params');
 };
 
-AnalysisFactory.prototype._onAnalysisRemoved = function (analysis) {
+AnalysisService.prototype._onAnalysisRemoved = function (analysis) {
   this._removeAnalsyisFromIndex(analysis);
   analysis.unbind('destroy', this._onAnalysisRemoved);
 };
 
-AnalysisFactory.prototype.findNodeById = function (id) {
+AnalysisService.prototype.findNodeById = function (id) {
   return this._analysisCollection.get(id);
 };
 
-AnalysisFactory.prototype._addAnalysisToCollection = function (analysis) {
+AnalysisService.prototype._addAnalysisToCollection = function (analysis) {
   return this._analysisCollection.add(analysis);
 };
 
-AnalysisFactory.prototype._removeAnalsyisFromIndex = function (analysis) {
+AnalysisService.prototype._removeAnalsyisFromIndex = function (analysis) {
   return this._analysisCollection.remove(analysis);
 };
 
-module.exports = AnalysisFactory;
+/**
+ * Return a list with all the analyses contained in the given collections.
+ */
+AnalysisService.getAnalysisList = function (layersCollection, dataviewsCollection) {
+  var layerAnalyses = _getAnalysesFromLayers(layersCollection);
+  var dataviewsAnalyses = _getAnalysesFromDataviews(dataviewsCollection);
+  return layerAnalyses.concat(dataviewsAnalyses);
+};
+
+function _getAnalysesFromLayers (layersCollection) {
+  var layers = _getCartoDBAndTorqueLayers(layersCollection);
+  return layers.map(function (layer) {
+    if (!layer.getSource()) {
+      throw new TypeError('CartoDB and Torque layers must have a source');
+    }
+    return layer.getSource();
+  });
+}
+
+function _getAnalysesFromDataviews (dataviewsCollection) {
+  return dataviewsCollection.map(function (dataview) {
+    return dataview.getSource();
```

Comment:
I think we need to check if getSource is null here too

Reply:
cc @alonsogarciapablo

Reply:
Or we could assume both layers and dataviews will always have a source [here](https://github.com/CartoDB/cartodb.js/pull/1795/files#diff-b4b64c4fcc702683c1df5ba40b85006bR107).